src/cryptography/hazmat/primitives/serialization/ssh.py

```diff
@@ -582,6 +588,59 @@ def encode_private(
         f_priv.put_sshstr(f_keypair)
 
 
+class _SSHFormatEd448:
+    """Format for Ed448 keys.
+
+    Public:
+        bytes point
+    Private:
+        bytes point
+        bytes secret_and_point
+    """
+
+    def get_public(
+        self, data: memoryview
+    ) -> tuple[tuple[memoryview], memoryview]:
+        """Ed448 public fields"""
+        point, data = _get_sshstr(data)
+        return (point,), data
+
+    def load_public(
+        self, data: memoryview
+    ) -> tuple[ed448.Ed448PublicKey, memoryview]:
+        """Make Ed448 public key from data."""
+        (point,), data = self.get_public(data)
+        public_key = ed448.Ed448PublicKey.from_public_bytes(
+            point.tobytes()
+        )
+        return public_key, data
+
+    def load_private(
+        self, data: memoryview, pubfields
+    ) -> tuple[ed448.Ed448PrivateKey, memoryview]:
+        """Make Ed448 private key from data."""
+        raise UnsupportedAlgorithm(
+            "Loading Ed448 SSH private keys is unsupported"
+        )
+
+    def encode_public(
+        self, public_key: ed448.Ed448PublicKey, f_pub: _FragList
+    ) -> None:
+        """Write Ed448 public key"""
+        raw_public_key = public_key.public_bytes(
+            Encoding.Raw, PublicFormat.Raw
+        )
+        f_pub.put_sshstr(raw_public_key)
+
+    def encode_private(
+        self, private_key: ed448.Ed448PrivateKey, f_priv: _FragList
+    ) -> None:
+        """Write Ed448 private key"""
+        raise UnsupportedAlgorithm(
+            "Serializing Ed448 SSH private keys is unsupported"
+        )
```

Comment:
I think it's *quite likely* that Ed448 private keys for OpenSSH would (will?) have a format very similar to Ed25519 private keys (e.g. https://github.com/pyca/cryptography/blob/d3eda718f1fc9fc5510bfb80127f7f2649cbfea2/tests/hazmat/primitives/test_ssh.py#L520-L530), but since I haven't seen any real-world examples of private keys *and* since the Ed25519 format is weirdly redundant in terms of its repetition of private and public key parts… I'm reluctant to make any assumptions about it here.

Reply:
On the other hand, from https://github.com/github/putty/commit/a085acbadf829ac5b426323ca98058d6aa4048ba …

> since the OpenSSH agent protocol is the de facto
standard … OpenSSH themselves get to
say what the format for a key should or shouldn't be. So if they don't
support a particular key method, what do you do?
>
> I checked with them, and they agreed that there's an obviously right
format for Ed448 keys, which is to ***do them exactly like Ed25519 except
that you have a 57-byte string everywhere Ed25519 had a 32-byte
string***. So I've done that.

So we could follow suit and support an OpenSSH-like Ed448 _private_ key format just as Putty does.